packages/block-editor/src/store/reducer.js

```diff
@@ -500,8 +504,18 @@ const withBlockReset = ( reducer ) => ( state, action ) => {
 				...mapBlockParents( action.blocks ),
 			},
 			cache: {
-				...omit( state.cache, visibleClientIds ),
-				...mapValues( flattenBlocks( action.blocks ), () => ( {} ) ),
+				...state.cache,
```

Comment:
Just curious why these were being `omit`-ed from everything in this reducer in the first place?  Now that we are no longer omitting them in the cache, does this raise any potential issues?

Reply:
I think it was just to keep it simpler.

Do you see an issue with this @aduth, @youknowriad?

Reply:
This has changed quite a bit since last I recall being familiar with it, but my concerns would be one of (a) are we risking to leave orphaned child blocks or (b) are we not cleaning up blocks which are no longer part of state.

Reading the documentation for the function, it seems like ideally we would want to avoid a merge altogether, and that the only reason we have it is to account for reusable blocks. I'm not familiar if we're to the point with recent changes like #14367 to be able to remove this and let `RESET_BLOCKS` entirely take over the state (no merge). But if not, then I think we probably still want the `omit` ? Since otherwise, we may leave some state for blocks which are no longer relevant for the next state, right? When is the cache ever cleaned up?

Reply:
>I'm not familiar if we're to the point with recent changes like #14367 to be able to remove this and let RESET_BLOCKS entirely take over the state (no merge).

That would be ideal, who would be a good person to ask?